### ISSUE TITLE
[2019/12/26] Machine Learning 輪講

### ISSUE THREAD

**Issue by agatan** [wantedly/machine-learning-round-table#31]
## Why

Machine Learning 輪講は最新の技術や論文を追うことで、エンジニアが「技術で解決できること」のレベルをあげていくことを目的にした会です。

prev. #27 

## What

話したいことがある人はここにコメントしましょう！
面白いものを見つけた時点でとりあえず話すという宣言だけでもしましょう！

**Comment by agatan** [comment]
先週に引き続き僕は参加できないです... skip かも。

**Comment by agatan** [comment]
## Reformer: The Efficient Transformer

- https://openreview.net/forum?id=rkgNKkHtvB
    - ICLR 2020 accept されたっぽい

Transformer はめちゃ流行ってるし性能いいんだけど、メモリ消費がつらい。
attention weights 計算のために `L ^ 2` だけ計算が必要になるため、 L が長くなるほど辛い。
（元祖 BERT pretrained model も 512 までくらいで切られていた気がする？）
また、層を増すごとに  gradient 計算のためにとっておかないといけない activation が増え、これもメモリ消費量増大に寄与している。

この論文では以下の 2 つを提案している

- Locality Sensitive Hashing を用いて attention weights の計算を省力化する LS Attention 
- Reversible Residual layer （RevNet 的なやつ）を用いることで backprop のために activation をとっておくのをやめる

実験がなんか不思議なことをやっていてよくわからなかった...

**Comment by hakubishin3** [comment]
## mecab-python3ってどうなったんだっけ
https://qiita.com/hi-asano/items/aaf406db875f1c81530e

**Comment by agatan** [comment]
### tf.keras に EfficientNet シリーズが仲間入り

https://github.com/tensorflow/tensorflow/commit/0d7620c3bc50b01054804d8c631b7eed45a7925b

**Comment by agatan** [comment]
### Transformers 2.3.0

https://github.com/huggingface/transformers/releases/tag/v2.3.0

- spacy みたいな Light-weight API が生えた。 
- `transformers-cli serve --task question-answering` とかができる CLI 
- tfhub, torchhub みたいなモデル共有のためのしくみ
- Japanese BERT や T5 などが追加

**Comment by yu-ya4** [comment]
## Optimally Balancing Receiver and Recommended Users’
Importance in Reciprocal Recommender Systems

http://www.inf.unibz.it/~ricci/papers/p131-kleinerman.pdf

- Reciprocal Recommender Systems シリーズ
- RecSys2018
- イイネする側の嗜好とイイネに対して返信する側の嗜好を Aggregate する際に重み付けをした。
- イスラエルのデーティングサービスでちゃんとオンラインテストした。
- 総マッチ数は落ちるけど，マッチ率は高まる。
